src/swap/SwapScreen.tsx

```diff
@@ -536,21 +536,27 @@ export function SwapScreen({ route }: Props) {
               onPressCta={onPressLearnMoreFees}
             />
           )}
-          {showPriceImpactWarning && (
-            <InLineNotification
-              severity={Severity.Warning}
-              title={t('swapScreen.priceImpactWarning.title')}
-              description={t('swapScreen.priceImpactWarning.body')}
-              style={styles.warning}
-            />
-          )}
-          {showMissingPriceImpactWarning && (
-            <InLineNotification
-              severity={Severity.Warning}
-              title={t('swapScreen.missingSwapImpactWarning.title')}
-              description={t('swapScreen.missingSwapImpactWarning.body')}
-              style={styles.warning}
-            />
+
+          {!exchangeRateUpdatePending && (
+            <>
+              {showPriceImpactWarning ? (
+                <InLineNotification
+                  severity={Severity.Warning}
+                  title={t('swapScreen.priceImpactWarning.title')}
+                  description={t('swapScreen.priceImpactWarning.body')}
+                  style={styles.warning}
+                />
+              ) : (
+                showMissingPriceImpactWarning && (
+                  <InLineNotification
+                    severity={Severity.Warning}
+                    title={t('swapScreen.missingSwapImpactWarning.title')}
+                    description={t('swapScreen.missingSwapImpactWarning.body')}
+                    style={styles.warning}
+                  />
+                )
+              )}
+            </>
```

Comment:
nit: then when `showPriceImpactWarning` or `showMissingPriceImpactWarning` is true it doesn't necessarily mean they should really be shown. Maybe this logic should be included directly when setting those variables?

Reply:
yeah you're right...i found the conditions were getting long and confusing to read, so thought it might be clearer to have the container control what happens during loading / the overall hierarchy. but indeed the tradeoff is that the logic is split :/ i moved the logic back to the variables themselves